context/app/static/js/components/entity-search/MetadataMenu/MetadataMenu.jsx

```diff
@@ -38,16 +39,28 @@ function NotebookMenuItem(props) {
   );
 }
 
-function MetadataMenu({ entityType }) {
+function MetadataMenu({ entityType, results }) {
   const lcPluralType = `${entityType.toLowerCase()}s`;
   const { selectedHits, createNotebook, closeMenu } = useMetadataMenu(lcPluralType);
   const menuID = 'metadata-menu';
 
   const { selectedRows } = useStore();
-  const errorMessage =
-    selectedRows.size > 10
-      ? `You have selected ${selectedRows.size} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`
-      : null;
+  const selectedRowsArray = useDatasetsAccessLevel(selectedRows.size > 0 ? [...selectedRows] : []).datasets;
```

Comment:
Could you clarify the use of the spread operator here? `useDatasetsAccessLevel` doesn't seem to modify the passed array in any way so I'm not sure what benefit cloning it brings.

Reply:
Sure! Selected Rows is a set, which is an object, so I need to convert it to an array as useDatasetsAccessLevel takes in an array as an argument. By using the square brackets, I am creating an array, and spreading the contents of the set into it.

Reply:
That makes sense, thanks for clarifying! I missed that `selectedRows` was a set (I should've been able to tell by the use of `.size` 😄)